### PR TITLE
Fix: search result page refreshing all items

### PR DESCRIPTION
**Pull request by JuliusKryger**
This pull request includes several changes to the frontend codebase, focusing on simplifying the search functionality and improving pagination handling. The most important changes include modifications to the `Pagination` component, `MediaDetailPage`, `SearchPage`, and related services.

Improvements to search functionality:

* [`src/frontend/src/pages/MediaDetailPage.jsx`](diffhunk://#diff-9067da5199ca855ad6e34c7c96e7704c01a34040a5826da405606ca2edcb83ecL22-R22): Removed the use of `fetchAllPages` for fetching titles and adjusted the dependency array in the `useEffect` hook to avoid infinite loops. [[1]](diffhunk://#diff-9067da5199ca855ad6e34c7c96e7704c01a34040a5826da405606ca2edcb83ecL22-R22) [[2]](diffhunk://#diff-9067da5199ca855ad6e34c7c96e7704c01a34040a5826da405606ca2edcb83ecL104-R104) [[3]](diffhunk://#diff-9067da5199ca855ad6e34c7c96e7704c01a34040a5826da405606ca2edcb83ecL123-R120)

* [`src/frontend/src/pages/SearchPage.jsx`](diffhunk://#diff-79dd884e96eabdac42ed88178084985c46779881f106d490811c733437081019L2-L6): Removed the filter functionality and refactored the search logic into separate functions for media and persons, improving code readability and maintainability. [[1]](diffhunk://#diff-79dd884e96eabdac42ed88178084985c46779881f106d490811c733437081019L2-L6) [[2]](diffhunk://#diff-79dd884e96eabdac42ed88178084985c46779881f106d490811c733437081019L17-R32) [[3]](diffhunk://#diff-79dd884e96eabdac42ed88178084985c46779881f106d490811c733437081019L46-R123) [[4]](diffhunk://#diff-79dd884e96eabdac42ed88178084985c46779881f106d490811c733437081019R133-R173)

Pagination handling improvements:

* [`src/frontend/src/components/Pagination/Pagination.jsx`](diffhunk://#diff-91b141d548ef43421e370c6d79e71519634d498810a79ffe97811a095b7d7039L12-R12): Added a check to prevent redundant page changes when the current page is selected again.

Service updates:

* [`src/frontend/src/services/mediaService.js`](diffhunk://#diff-337b1594ea5513a006c2589f5ce02e2b7ba67fdb8be37b75ced9be883c1b4dcfL19-L24): Simplified the `fetchTitles` function by removing unnecessary pagination parameters.

* [`src/frontend/src/services/personService.js`](diffhunk://#diff-c4a26b3d097e249c6cc68e5d3d2110a8148b3fd1495828acbe2cb8d682070142L25-R31): Updated the `fetchPersons` function to accept a single object parameter for better clarity and flexibility.

### Other Fixes:

- Removed the spinner component.
- Resolved issues with refreshing the page during pagination.

### CODE REVIEW

File: src/frontend/src/pages/SearchPage.jsx [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import { useState, useEffect, useCallback } from 'react';
-import { Container, Spinner } from 'react-bootstrap';
+import { Container } from 'react-bootstrap';
```
Comment (idax6797) [inline src/frontend/src/pages/SearchPage.jsx:2]: Fine with me to remove it

File: src/frontend/src/services/personService.js [hunk 2]
```diff
@@ -22,14 +22,16 @@ const enhancePersonWithImage = async (person, imgSize = ImageSize.Normal) => {
...
-export const fetchPersons = async (filter, page, count) => {
+export const fetchPersons = async (query) => {
   const api = new ApiClient();
+  const queryParams = [];
   try {
-    const queryParams = [];
-    if (page) queryParams.push({ key: 'page', value: page });
-    if (count) queryParams.push({ key: 'count', value: count });
-    if (filter?.name) queryParams.push({ key: 'name', value: filter.name });
+    console.log('Query:', query);
```
Comment (idax6797) [inline src/frontend/src/services/personService.js:29]: remove console.log

File: src/frontend/src/services/personService.js [hunk 3]
```diff
@@ -22,14 +22,16 @@ const enhancePersonWithImage = async (person, imgSize = ImageSize.Normal) => {
...
+    if (query) queryParams.push({ key: 'page', value: query.page });
+    if (query) queryParams.push({ key: 'count', value: query.count });
+    if (query) queryParams.push({ key: 'name', value: query.name });
 
+    console.log('Query params:', queryParams);
```
Comment (idax6797) [inline src/frontend/src/services/personService.js:34]: Remove console.log

**Review by idax6797 — APPROVED** [review]
Fine with not having the spinner, it looks to work fine

**Review by idax6797 — CHANGES_REQUESTED** [review]
Also this could be implemented in this draft

https://github.com/orgs/RUC-MSc-CS-CIT-2024/projects/4/views/1?pane=issue&itemId=90958797&issue=RUC-MSc-CS-CIT-2024%7Cportfolio_subproject_3%7C150

**Review by idax6797 — CHANGES_REQUESTED** [review]
Nice it has been removed, but when I search after eg Jennifer Aniston. Then the person result now does not show

**Comment by JuliusKryger** [comment]
That is a backend issue, if you search with capital letters, a search should be done with lower case letters.

File: src/frontend/src/pages/SearchPage.jsx [hunk 4]
```diff
@@ -43,85 +42,66 @@ export default function SearchPage() {
...
+        name: query_data.query,
+        page: personPage.page,
+        count: personPage.count,
+      };
 
-        if (params.query_type !== 'Structured') {
-          let q = params.query;
-          if (params.query_type !== 'Simple') {
-            q = params.keywords.join(' ');
-          }
```
Comment (sofushn) [inline src/frontend/src/pages/SearchPage.jsx:87]: Person query should only happen if not structured search, maybe add an `if` statement to the effect to ensure its not called if structured search, and is cleared instead

File: src/frontend/src/services/personService.js [hunk 5]
```diff
@@ -22,13 +22,13 @@ const enhancePersonWithImage = async (person, imgSize = ImageSize.Normal) => {
   return person;
 };
 
-export const fetchPersons = async (filter, page, count) => {
+export const fetchPersons = async (query) => {
```
Comment (sofushn) [inline src/frontend/src/services/personService.js:25]: Do like other service functions and use `{}` as the argument

```suggestion
export const fetchPersons = async ({page, count, name}) => {
```

File: src/frontend/src/services/personService.js [hunk 6]
```diff
@@ -22,13 +22,13 @@ const enhancePersonWithImage = async (person, imgSize = ImageSize.Normal) => {
...
   const api = new ApiClient();
+  const queryParams = [];
   try {
-    const queryParams = [];
-    if (page) queryParams.push({ key: 'page', value: page });
-    if (count) queryParams.push({ key: 'count', value: count });
-    if (filter?.name) queryParams.push({ key: 'name', value: filter.name });
+    if (query) queryParams.push({ key: 'page', value: query.page });
+    if (query) queryParams.push({ key: 'count', value: query.count });
+    if (query) queryParams.push({ key: 'name', value: query.name });
```
Comment (sofushn) [inline src/frontend/src/services/personService.js:31]:
```suggestion
    if (query) queryParams.push({ key: 'page', value: page });
    if (query) queryParams.push({ key: 'count', value: count });
    if (query) queryParams.push({ key: 'name', value: name });
```

**Review by sofushn — CHANGES_REQUESTED** [review]

**Comment by idax6797** [comment]
Screenshot 2024-12-16 at 14 36 01


There is still no titles and there is a error or something weird is happening here

**Comment by idax6797** [comment]
Screenshot 2024-12-16 at 14 38 48


And the person on best match jennifer aniston is not relevant, can that be fixed

**Comment by JuliusKryger** [comment]
https://github.com/RUC-MSc-CS-CIT-2024/portfolio_subproject_3/pull/154?new_mergebox=false#issuecomment-2545660816
image

**Review by sofushn — CHANGES_REQUESTED** [review]

File: src/frontend/src/utils/fetchAllPages.js [hunk 7]
```diff
@@ -11,8 +11,8 @@ export const fetchAllPages = async (
     const result = personId
       ? await fetchFunction(personId, currentPage, count)
       : await fetchFunction(currentPage, count);
-    allData = [...allData, ...result.items];
-    hasMoreData = result.items.length === count;
+    allData = [...allData, ...result];
+    hasMoreData = result.length === count;
```
[the remark this answers is not in the document]
Comment (idax6797) [reply src/frontend/src/utils/fetchAllPages.js:15]: Yes this is used in fetcing the bookmarks, completed and following, so please do set this up for these services also if removed

**Review by sofushn — APPROVED** [review]
Good stuff 🚀

**Review by idax6797 — APPROVED** [review]

**Pull request merged**